operator.js

```diff
@@ -0,0 +1,35 @@
+const operate = (operator, num1, num2) => {
+  let result;
+
+  if (typeof num1 !== 'number' || typeof num2 !== 'number') {
+    return 'Both operands must be numbers';
+  }
+
+  switch (operator) {
+    case '+':
+      result = num1 + num2;
+      break;
+    case '-':
+      result = num1 - num2;
+      break;
+    case '*':
+      result = num1 * num2;
+      break;
+    case '/':
+      if (num2 === 0) {
+        return 'Division by zero is not allowed';
+      }
+      result = num1 / num2;
+      break;
+    default:
+      return 'Invalid operator';
+  }
+  if (num1 >= 1e14 || num1 <= -1e14 || num2 >= 1e14 || num2 <= -1e14 || result > 1e14) {
+    return 'out of range';
+  }
```

Comment:
That a look at this, why don't we separate the operands range and add it immediately after your typeOf, I mean where you are checking the number, let add the range of both num1 and num2 there, and check the result range down here seperately